src/entity.ts

```diff
@@ -376,11 +437,45 @@ export namespace entity {
     return value instanceof entity.Key;
   }
 
+  /**
+   * Convert a protobuf `integerValue`.
+   *
+   * @private
+   * @param {object} value The `integerValue` to convert.
+   */
+  function decodeIntegerValue(value: ValueProto) {
+    const num = Number(value.integerValue);
+    if (!Number.isSafeInteger(num)) {
+      throw new Error(
+        'We attempted to return all of the numeric values, but ' +
+          (value.propertyName ? value.propertyName + ' ' : '') +
+          'value ' +
+          value.integerValue +
+          " is out of bounds of 'Number.MAX_SAFE_INTEGER'.\n" +
+          "Please consider passing 'options.wrapNumbers=true' and\n" +
+          "'options.integerTypeCastOptions' as\n" +
+          '{\n' +
+          '  integerTypeCastFunction: optionally provide <your_custom_function>\n' +
```

Comment:
I would lose `optionally` here from the description, because once you've opted in to `integerTypeCastOptions`, this is no longer optional.